src/cli/common.rs

```diff
@@ -491,22 +492,24 @@ fn print_toolchain_path(
 }
 
 pub(crate) fn list_toolchains(cfg: &Cfg, verbose: bool) -> Result<utils::ExitCode> {
-    // Work with LocalToolchainName to accommodate path based overrides
-    let toolchains = cfg
-        .list_toolchains()?
-        .iter()
-        .map(Into::into)
-        .collect::<Vec<_>>();
+    let toolchains: Vec<ToolchainName> = cfg.list_toolchains()?;
     if toolchains.is_empty() {
         writeln!(process().stdout().lock(), "no installed toolchains")?;
     } else {
-        let def_toolchain_name = cfg.get_default()?.map(|t| (&t).into());
+        let def_toolchain_name = cfg.get_default()?;
         let cwd = utils::current_dir()?;
-        let ovr_toolchain_name = if let Ok(Some((toolchain, _reason))) = cfg.find_override(&cwd) {
-            Some(toolchain)
-        } else {
-            None
-        };
+        let ovr_toolchain_name: Option<ToolchainName> =
+            if let Ok(Some((LocalToolchainName::Named(toolchain), reason))) =
+                cfg.find_active_toolchain(&cwd)
+            {
+                if !matches!(reason, crate::config::ActiveReason::Default) {
```

Comment:
Style nit: let's import `ActiveReason` here, and use a `match` expression` instead of `matches!()`. Also please avoid type annotations (here and elsewhere) -- I think this one is probably unnecessary.